### PR TITLE
EWL 545#2 Detecting Window Resize on ViewMore Component to update its height

### PR DESCRIPTION
**Pull request by saad-kh**
The ViewMore Component should have cared about the resizing of the Window because a new Window`size can make its displayed height less accurate.
This PR fix that by decorating it with an external [package ](https://github.com/maslianok/react-resize-detector) HOC that rerenders the ViewMore component on window resize.

### CODE REVIEW

**Comment by arturkin** [comment]
Do we really need a new dependency instead of 
```
window.addEventListener(
   'resize', 
   throttle(()=> this.setState({windowWidth: window.innerWidth})), 500);
```

**Comment by saad-kh** [comment]
@arturkin  
You're right and that was the initial plan. However without the dependency we'd need to add and remove the event Listener (cleanup code), and make the setState call that doesn't have an explicit meaning but is here only for the rerendering side effect. The dependency let us reason about it with a more straightforward React paradigm.

**Comment by arturkin** [comment]
43Kb gzipped dependency with it's own deps for window.onresize it too much even in react word 😄 
http://maslianok.github.io/react-resize-detector/static/js/main.ef81d4ad.js

**Comment by saad-kh** [comment]
Wouw that's too much indeed. You're right I'll revert to a simple addEventListener.

**Comment by saad-kh** [comment]
@Xb0X @arturkin Updated

File: src/modules/ViewMore/index.js [hunk 1]
```diff
@@ -40,8 +40,15 @@ export default class ViewMore extends React.Component {
     // It is better to set the default value of expanded after the initial mount.
     // Otherwise the first render will not be accurate as the listRef might not be set yet.
     this.setState({ expanded: false });
+    window.addEventListener('resize', this.handleWindowResize);
```
Comment (arturkin) [inline src/modules/ViewMore/index.js]: Please add throttle/debounce

File: src/modules/ViewMore/index.js [hunk 2]
```diff
@@ -40,8 +40,15 @@ export default class ViewMore extends React.Component {
...
   }
 
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.handleWindowResize);
+  }
+
+  handleWindowResize = () => this.setState({});
```
Comment (arturkin) [inline src/modules/ViewMore/index.js]: You will loose `expanded` state here?

Comment (saad-kh) [reply src/modules/ViewMore/index.js]: No it'll preserve the current state.

**Comment by saad-kh** [comment]
@arturkin Throttle added

**Pull request merged**